liquibase-core/src/main/java/liquibase/database/core/PostgresDatabase.java

```diff
@@ -330,9 +338,22 @@ protected SqlStatement getConnectionSchemaNameCallStatement() {
         return new RawCallStatement("select current_schema()");
     }
 
+    /**
+     * Generates PK following {@code PostgreSQL} conventions:
+     * <li>Postgres PK size is limited with 63 bytes.
+     * <li>Postgres PK is suffixed with '_pkey'.
+     *
+     * @param tableName Table name as the base name for the generated PK.
+     * @return PK name.
+     */
     @Override
     public String generatePrimaryKeyName(final String tableName) {
-        return tableName.toUpperCase(Locale.US) + "_PKEY";
+        final byte[] bytes = tableName.getBytes(StandardCharsets.UTF_8);
+        final int suffixBytes = PGSQL_PK_SUFFIX.length(); // All characters are ASCII symbols => 1 byte per symbol.
+
+        return bytes.length <= PGSQL_PK_BYTES_LIMIT - suffixBytes
```

Comment:
I'd declare a variable for the max base name length here:

`final int maxBaseLen = PGSQL_PK_BYTES_LIMIT - suffixBytes;`

And use it in the two operations below.

Reply:
Not required, but still a good suggestion.